Add sample nagios service implementation

miguelxpn opened the pull request.

Commit:
Add sample nagios service implementation

ArSn commented on nagios-service/README.md:
```diff
@@ -0,0 +1,11 @@
+## Setup
+- Run nagios using docker: `docker run --name nagios4 -p 0.0.0.0:8080:80 jasonrivers/nagios:latest`
+- Open a shell session on your container and install ruby and wpscan:
+    - `docker exec -it nagios4 bash`
+    - `apt update`
+    - `apt install ruby-full`
+    - `gem install wpscan`
```
Nitpick: What do you think about making this one step instead of 4?

```suggestion
- Open a shell session on your container and install ruby and wpscan: `docker exec -it nagios4 /bin/sh -c "apt update && apt install -y ruby-full && gem install wpscan"`
```

miguelxpn replied:
Much better!

ArSn commented on nagios-service/check_wpscan.sh:
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Hostname argument
+hostname="$1"
```
We're not using this. I think it was intended for line 10? Otherwise we can just remove it here

miguelxpn replied:
D'oh, yeah, it was intended for line 10 to make is more readable, but then I just used the first argument directly there. I'll adjust.

ArSn commented on nagios-service/check_wpscan.sh:
```diff
@@ -0,0 +1,27 @@
+  exit 1
+fi
+
+if [ "$return_code" -gt 1 ]; then
+  echo "CRITICAL: $return_code vulnerabilities found."
+  exit 2
+else
```
are we certain about the return code being the number of vulns? I've tested this against a fresh JN site and I'm getting `[07-07-2023 12:59:01] SERVICE ALERT: <my-hostname>;WPScan;CRITICAL;SOFT;2;CRITICAL: 3 vulnerabilities found.` -- but when I run the same command manually it detects zero vulns? 🤔

miguelxpn replied:
I remembered reading an issue that they were considering returning the number of vulns as the exit code, I just looked at the code and they just return an indicator if it's vulnerable or not, I'll do some `jq` processing to see the output instead.

ArSn reviewed (changes requested):
This does not report the correct number of vulns or I am using it wrong (see other comment). Please advise 🙂

ArSn commented:
Oh I forgot to add: Could you please also add this example to the [list of examples in the main README.md file](https://github.com/Automattic/wpscan-examples#table-of-projects)?

miguelxpn committed:
Improve install wpscan instructions

Co-authored-by: Kolja Zuelsdorf <koljaz@web.de>

Commit:
Update main readme and use JQ to report the correct number of vulns

ArSn reviewed (changes requested):
As discussed, it seems we need to find a way to run the wpscan command under the correct nagios user and add documentation for that here.

Commit:
Add sudo instructions and timeout config increase

miguelxpn commented:
My experiments with Docker weren't successful, for some reason when running inside Nagios Docker would just fail, by running on the shell with the same nagios user it wouldn't, couldn't find out why so I changed the instructions to set up sudo.

I also added an instruction to increase the timeout configuration as sometimes the wpscan command might take more than a minute to run.

ArSn reviewed (approved):
All works as expected now 👍🏻
